lec09-modeling-r.Rmd

```diff
@@ -1,4 +1,615 @@
+
+# x = 0, y = 0 
+points(0, 0, # x, y coordinates of point to plot
+       cex = 1.5) # 'cex' stands for 'character expansion': marker size relative to default
+
+# x = ad/bc, y = a/b 
+x_star <- a * d / (b * c)
+y_star <- a / b
+points(x_star, y_star, # x, y coordinates of point to plot
+       pch = 21, # pch = 21 gives a filled circle
+       # see  https://stat.ethz.ch/R-manual/R-devel/library/graphics/html/points.html
+       bg = 'black', # background colour
+       cex = 1.5) 
+```
+
+One interesting thing about this system is that $x = 0, y = 0$ is an unstable fixed point, but
+there is one direction in which trajectories are attracted to it: along the y-axis. This makes
+sense: if there are no prey ($x = 0$), then the predators must also go extinct. We will come
+back to this later when we talk about analyzing stability analytically.
+
+## Null clines in two-dimensional systems
+
+A null cline is a curve on which one of derivatives in the model is $0$. In other words, 
+it's a series of points for which one of the variables is not changing --- its rate of change
+is $0$ on that line. 
+
+Finding null clines is similar to finding fixed points, except that when we found fixed points,
+we required both derivatives to be *simultaneously* zero. For null clines, we only set one
+derivative to zero at a time, and the other one is free to do what it wants. 
+
+To find the $x$ null cline(s), the curve(s) on which $x$ will not change, we set 
+$\frac{dx}{dt} = ax - bxy$ to zero, ignoring $\frac{dy}{dt}$. 
+
+$$0 = ax - bxy$$
+
+Now we proceed exactly as if we were finding the fixed points, but we don't require that $y$
+be anything in particular. We can factor the right hand side to $x(a-by)$, which means there
+are actually two curves for which the rate of change of $x$ is $0$. One of them is at $x = 0$:
+it's a straight line on the y-axis. For any value of $y$, provided $x = 0$, $x$ won't change.
+The other one is at $y = a/b$: another straight line, this time at a fixed value of $y$.
+For any value of $x$, provided $y = a/b$, $x$ won't change. 
+
+Let's run a numerical solution for the predator-prey model we've been using and plot the null
+clines we just found.
+
+```{r}
+# run the numerical solution
+library(deSolve)
+
+parameters = c(a = 5, b = 1, c = 1, d = 0.2) 
+state <- c(X = 2, Y = 2)
+times <- seq(0, 50, by = 0.01)
+
+result <- ode(y = state, times = times, func = predator_prey, parms = parameters)
+result <- data.frame(result)
+
+```
+
+Plot a trajectory in the phase plane and the null clines:
+
+```{r}
+library(ggplot2)
+
+x_seq <- seq(0, 8, by = 0.1)
+y_seq <- seq(0, 8, by = 0.1)
+
+df <- data.frame(x_seq, y_seq)
+
+p1 <- ggplot() + # assign plot to a variable so we can use it later
+  geom_path(aes(x = X, y = Y, color = time), result) + # trajectory
+  geom_line(aes(x = 0, y = y_seq), df, color = 'dark red') + # x null cline 1
+  geom_line(aes(x = x_seq, y = a/b), df, color = 'dark red') # x null cline 2
+
+p1
```

Comment:
Is understand both lines analytically, but only the y=5 line from the graph... The system seem to never touch the x=0 line, is that just a coincidence based on the values we chose?

Reply:
The x = 0 line is sort of special in that it's a stable direction - if you scroll up two plots you can see that trajectories that start on that line stay on it and go to x = 0, y = 0.